grades/api.py

```diff
@@ -55,8 +55,11 @@ def _compute_grade_for_non_fa(user_edx_run_data):
         UserFinalGrade: a namedtuple of (float, bool,) representing the final grade
             of the user in the course run and whether she passed it
     """
-    run_passed = user_edx_run_data.certificate is not None
-    grade = user_edx_run_data.certificate.grade if run_passed else user_edx_run_data.current_grade.percent
+    run_passed = user_edx_run_data.certificate is not None and user_edx_run_data.certificate.status == 'downloadable'
```

Comment:
Nitpick: I would suggest breaking this into multiple lines to make it easier to read.

```python
run_passed = (
    user_edx_run_data.certificate is not None and
    user_edx_run_data.certificate.status == 'downloadable'
)
```